simplecpp.cpp

```diff
@@ -1282,12 +1270,99 @@ void simplecpp::TokenList::constFoldBitwise(Token *tok)
                 result = (stringToLL(tok->previous->str()) ^ stringToLL(tok->next->str()));
             else /*if (*op == '|')*/
                 result = (stringToLL(tok->previous->str()) | stringToLL(tok->next->str()));
-            tok = tok->previous;
-            tok->setstr(toString(result));
-            deleteToken(tok->next);
-            deleteToken(tok->next);
+            simpleSquash(tok, toString(result));
+        }
+    }
+}
+
+static simplecpp::Token * stepForward(simplecpp::Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    return tok->next;
+}
+
+static simplecpp::Token * stepBack(simplecpp::Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    return tok->previous;
+}
+
+static int getTokensDeleteCount(simplecpp::Token *tok, const std::set<std::string> & breakPoints, simplecpp::Token *(step)(simplecpp::Token *), const std::pair<std::string, std::string> & brackets)
+{
+    int count = 0;
+    bool skip = false;
+    tok = (step)(tok);
+    for (; tok; tok = (step)(tok)) {
+        if (skip){
+            ++count;
+            if (tok->str() == brackets.second) {
+                skip = false;
+                continue;
+            }
+        } else if (tok->str() == brackets.first) {
+            skip = true;
+            ++count;
+        } else if (breakPoints.count(tok->str()) != 0) {
+            break;
+        } else {
+            ++count;
         }
     }
+    return count;
+}
+
+void simplecpp::TokenList::simpleSquash(Token *&tok, const std::string & result)
+{
+    tok = tok->previous;
+    tok->setstr(result);
+    deleteToken(tok->next);
+    deleteToken(tok->next);
+}
+
+void simplecpp::TokenList::squashTokens(Token *&tok, const std::set<std::string> & breakPoints, bool forwardDirection, const std::string & result)
+{
+    Token *(*step)(Token *) = &stepForward;
+    std::pair<std::string, std::string> brackets = std::make_pair<std::string, std::string>("(", ")");
```

Comment:
I am not sure that we should pass `brackets` as a parameter. It's redundant info if we pass direction.